src/content/encoding.rs

```diff
@@ -0,0 +1,58 @@
+use crate::headers::HeaderValue;
+use std::fmt::{self, Display};
+
+/// Available compression algorithms.
+#[non_exhaustive]
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum Encoding {
```

Comment:
It may be useful for this to have a `PartialEq<&str>`

e.g. `assert_eq!(res[headers::CONTENT_ENCODING], Encoding::Deflate);`